src/cpuinfo/SDL_cpuinfo.c

```diff
@@ -1044,42 +1049,48 @@ int SDL_GetSystemRAM(void)
             Uint64 memsize = 0;
             size_t len = sizeof(memsize);
 
-            if (sysctl(mib, 2, &memsize, &len, NULL, 0) == 0) {
+            if (sysctl(mib, 2, &memsize, &len, NULL, 0) == -1) {
+                SDL_SetError("sysctl");
+            } else {
                 SDL_SystemRAM = (int)(memsize / (1024 * 1024));
             }
         }
-#endif
-#if defined(__WIN32__) || defined(__GDK__)
-        if (SDL_SystemRAM <= 0) {
+#elif defined(__WIN32__) || defined(__GDK__)
+        {
             MEMORYSTATUSEX stat;
             stat.dwLength = sizeof(stat);
-            if (GlobalMemoryStatusEx(&stat)) {
+            if (!GlobalMemoryStatusEx(&stat)) {
+                SDL_SetError("GlobalMemoryStatusEx failed, error=%08X", (unsigned int)GetLastError());
+            } else {
                 SDL_SystemRAM = (int)(stat.ullTotalPhys / (1024 * 1024));
             }
         }
-#endif
-#ifdef __RISCOS__
-        if (SDL_SystemRAM <= 0) {
+#elif defined(__RISCOS__)
+        {
             _kernel_swi_regs regs;
             regs.r[0] = 0x108;
-            if (_kernel_swi(OS_Memory, &regs, &regs) == NULL) {
+            if (_kernel_swi(OS_Memory, &regs, &regs)) {
+                SDL_SetError("_kernel_swi");
```

Comment:
`_kernel_swi` returns a pointer to a `_kernel_oserror` struct, which can be used to provide a more detailed error message. For example:
```
_kernel_oserror *error;
_kernel_swi_regs regs;
regs.r[0] = 0x108;
error = _kernel_swi(OS_Memory, &regs, &regs);
if (error != NULL) {
    SDL_SetError("OS_Memory 8 failed: %s (%i)", error->errmess, error->errnum);
}
```

Reply:
@madebr, did you see this feedback?